### PR TITLE
Migrate demo movies to es6 

### PR DESCRIPTION
**Pull request by sunnylqm**

### CODE REVIEW

**Comment by sunnylqm** [comment]
@corbt Git is really complicated for me...

**Comment by mkonicek** [comment]
Thanks for doing this! Can you provide a Test Plan, ideally a few screenshots showing the app running? It's just a standard thing at fb to provide a Test Plan explaining how you tested a change with every commit.

**Comment by sunnylqm** [comment]
Any Test Plan example that I can learn?

**Comment by corbt** [comment]
@sunnylqm it sounds like you just need to include a couple of screen shots here in the PR demonstrating that the example movie app still runs correctly after these changes have been applied. Thanks!

**Comment by sunnylqm** [comment]
simulator screen shot feb 11 2016 09 53 36
simulator screen shot feb 11 2016 09 53 49
simulator screen shot feb 11 2016 09 54 28

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
This fails some tests in `Examples/Movies/__tests__/getImageSource-test.js:12:36`.

**Comment by sunnylqm** [comment]
image

**Comment by sunnylqm** [comment]
Update test code according to this: https://github.com/facebook/jest/issues/666

Now it passed.
screen shot 2016-02-12 at 01 03 15

**Comment by sunnylqm** [comment]
@mkonicek How about now?

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Comment by bestander** [comment]
I see this error in e2e tests:
```
Test runner detected test failures. To see all failures go to: https://our.intern.facebook.com/intern/testinfra/testrun/117826733
//Libraries/FBReactKit:FBReactKitIntegrationTests - CatalystIntegrationTests/testMoviesApp: failed
Failure: failed: caught "NSInternalInconsistencyException", "*** Assertion failure in -[RCTTestRunner runTest:module:initialProps:configurationBlock:expectErrorBlock:], Libraries/FBReactKit/js/react-native-github/Libraries/RCTTest/RCTTestRunner.m:154: RedBox error: Unhandled JS Exception: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object. Check the render method of `Snapshotter`."

```

**Comment by sunnylqm** [comment]
@bestander which means?

**Comment by bestander** [comment]
No idea actually, some internal code got broken.
Just FYI that we will need to merge it manually and resolve the conflicts.

**Comment by mkonicek** [comment]
Those are screenshot tests. Not sure why the code is not on GitHub, someone has to find time to either open source those tests or manually merge this PR. It has the 'Import failed' label so we don't forget it.

**Comment by jsierles** [comment]
Have these e2e tests been open sourced yet?

**Comment by bestander** [comment]
Not yet :(
It is high on my list.
The reason is that we want to run those tests both internally at FB and
externally on CI.
I was waiting for the ability to run CI style tests in our internal build
infrastructure.
Because then I would not need to support tests execution for two completely
different environments.

This ability is finally in testing phase, after we roll this out and it is
stable it would take very little effort to finally move the tests to open
source.

**Comment by mkonicek** [comment]
@bestander Is now working on open sourcing those tests \o/

File: Examples/Movies/MovieCell.js [hunk 1]
```diff
@@ -15,23 +15,23 @@
  */
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
```
Comment (stereodenis) [inline Examples/Movies/MovieCell.js]: Now it is in 'react' package

Comment (satya164) [inline Examples/Movies/MovieCell.js]: e.g. `import React, { Component } from 'react';`

File: Examples/Movies/MovieScreen.js [hunk 2]
```diff
@@ -15,21 +15,21 @@
  */
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
```
Comment (satya164) [inline Examples/Movies/MovieScreen.js]: `import React, { Component } from 'react';`

File: Examples/Movies/MoviesApp.android.js [hunk 3]
```diff
@@ -16,18 +16,18 @@
  */
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
```
Comment (satya164) [inline Examples/Movies/MoviesApp.android.js]: `import React, { Component } from 'react';`

File: Examples/Movies/MoviesApp.ios.js [hunk 4]
```diff
@@ -16,17 +16,17 @@
  */
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
```
Comment (satya164) [inline Examples/Movies/MoviesApp.ios.js]: `import React, { Component } from 'react';`

File: Examples/Movies/SearchBar.android.js [hunk 5]
```diff
@@ -16,23 +16,24 @@
  */
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
```
Comment (satya164) [inline Examples/Movies/SearchBar.android.js]: `import React, { Component } from 'react';`

File: Examples/Movies/SearchBar.ios.js [hunk 6]
```diff
@@ -16,16 +16,16 @@
  */
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
```
Comment (satya164) [inline Examples/Movies/SearchBar.ios.js]: `import React, { Component } from 'react';`

File: Examples/Movies/SearchScreen.js [hunk 7]
```diff
@@ -15,32 +15,31 @@
  */
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
```
Comment (satya164) [inline Examples/Movies/SearchScreen.js]: `import React, { Component } from 'react';`

File: Examples/Movies/__tests__/getImageSource-test.js [hunk 8]
```diff
@@ -5,7 +5,8 @@
 
 jest.dontMock('../getImageSource');
 
-var getImageSource = require('../getImageSource');
+//see https://github.com/facebook/jest/issues/666
+const getImageSource = require('../getImageSource').default;
```
Comment (satya164) [inline Examples/Movies/__tests__/getImageSource-test.js]: as of latest jest, I think you can just do `import getImageSource from '../getImageSource';`

**Comment by facebook-github-bot-2** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @sebmarkbage as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by bestander** [comment]
@facebook-github-bot import

**Comment by facebook-github-bot-9** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/166149433762431/int_phab) to review internal test results.

**Comment by bestander** [comment]
@facebook-github-bot import

**Comment by facebook-github-bot-7** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/166149433762431/int_phab) to review internal test results.

**Comment by bestander** [comment]
@sunnylqm, if you want to give it another try :) could you rebase it?

**Comment by javache** [comment]
These have now all been migrate automatically. Thanks for your PR!

**Pull request closed without merging**